chia/rpc/wallet_rpc_api.py

```diff
@@ -260,6 +261,41 @@ async def delete_key(self, request):
             path.unlink()
         return {}
 
+    #
+    # Check the key for a few things
+    # 1) is it used for either farm or pool rewards
+    # 2) do any wallets have a non-zero balance
+    #
+    async def check_delete_key(self, request):
+        fingerprint = request["fingerprint"]
+        sk, _ = await self._get_private_key(fingerprint)
+
+        used_for_farmer, used_for_pool = check_key_used_for_rewards(self.service.root_path, sk, 100)
+
+        if self.service.logged_in_fingerprint != fingerprint:
+            await self.service._start(fingerprint=fingerprint, skip_backup_import=True)
+
+        walletBalance: bool = False
+        wallets: List[WalletInfo] = await self.service.wallet_state_manager.get_all_wallet_info_entries()
+
+        for w in wallets:
+            wallet = self.service.wallet_state_manager.wallets[w.id]
```

Comment:
Wallet can be deleted, so we should probably lock ouside get_wallet_info_entries

Reply:
code updated